File: CS/CustomFunctionCriteriaOperator.Module/BusinessObjects/ValidatedObject.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace CustomFunctionCriteriaOperator.Module.BusinessObjects {
   [DefaultClassOptions]
   [RuleCriteria("", DefaultContexts.Save, "Company.Oid = CurrentCompanyOid()")]
   public class ValidatedObject : BaseObject {
      public ValidatedObject(Session session)
         : base(session) {}
      public override void AfterConstruction() {
         base.AfterConstruction();
      }
      private Company company;
      public Company Company {
         get {
            return company;
         }
         set {
            SetPropertyValue("Company", ref company, value);
         }
      }
      private string name;
      public string Name {
          get {
              return name;
          }
          set {
              SetPropertyValue("Name", ref name, value);
          }
      }
   }

}

File: VB/CustomFunctionCriteriaOperator.Module/BusinessObjects/ValidatedObject.vb
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace CustomFunctionCriteriaOperator.Module.BusinessObjects
   <DefaultClassOptions, RuleCriteria("", DefaultContexts.Save, "Company.Oid = CurrentCompanyOid()")> _
   Public Class ValidatedObject
       Inherits BaseObject

      Public Sub New(ByVal session As Session)
          MyBase.New(session)
      End Sub
      Public Overrides Sub AfterConstruction()
         MyBase.AfterConstruction()
      End Sub

      Private company_Renamed As Company
      Public Property Company() As Company
         Get
            Return company_Renamed
         End Get
         Set(ByVal value As Company)
            SetPropertyValue("Company", company_Renamed, value)
         End Set
      End Property

      Private name_Renamed As String
      Public Property Name() As String
          Get
              Return name_Renamed
          End Get
          Set(ByVal value As String)
              SetPropertyValue("Name", name_Renamed, value)
          End Set
      End Property
   End Class

End Namespace
